src/inmanta/agent/in_process_executor.py

```diff
@@ -90,6 +112,8 @@ def join(self, thread_pool_finalizer: list[ThreadPoolExecutor]) -> None:
         assert self._stopped
         thread_pool_finalizer.append(self.provider_thread_pool)
         thread_pool_finalizer.append(self.thread_pool)
+        if self.periodic_cache_cleanup_job:
+            await self.periodic_cache_cleanup_job
```

Comment:
I believe this will suffer from the same issue as raised by @arnaudsjs [here](https://inmanta.slack.com/archives/CKRF0C8R3/p1721120710485599).